Add wip jiggle physics to swets

SollyW opened the pull request:
Makes swets absorb entities more nicely than forcing the entity to ride the swet. WIP; a few things need to be changed and tested, see TODOs in the code comments and #developers on discord

SollyW committed:
Add wip jiggle physics to swets

Jack-Papel committed:
Dampening

The original implementation was of a linear restoring force, meaning the player would oscillate back and forth like a spring. This now has added dampening

Jack-Papel commented:
Added some dampening, since the original code made the player spring back and forth quite vigorously. Now the player's sloshing will mellow. I didn't get rid of the sloshing entirely, since it's quite nice in small doses

Jack-Papel committed:
Fix white swet transformation & loosened larger swets pull

White swets can once again change color.
Larger swets have a looser pull on the player.

Jack-Papel commented:
White swets change color when they get different entities stuck in them. This is unique to Aether Reborn, so I get why you missed it. Fixed with the previous commit

Jack-Papel committed:
Pickup conditions

Introduces pickup conditions for swets.
Pets can't be picked up on Easy difficulty.
Non players-or-pets can't be picked up if MobGriefing is off.
If an entity cannot be picked up, it'll have regular knockback.

Jack-Papel commented:
Added pickup conditions. Will make more edits tomorrow. Feel free to add some more of your own in the meantime if you think of anything you or I have forgotten.

Jack-Papel committed:
Remove TODO, taken care of

Forgot to remove this TODO. #developers decided that pets shouldn't be picked up on easy mode, and that other non-player entities shouldn't be picked up if mobGriefing is set to false.

Jack-Papel committed:
Swets' pulls are affected by the size and number of entities they pull

Jack-Papel committed:
Remove TODO

Works on multiplayer (tested with LAN)
Also got rid of a print statement I left in.
